df1-html/lib/Df1/Html/Parse.hs

```diff
@@ -0,0 +1,88 @@
+{-# LANGUAGE OverloadedLists #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+
+module Df1.Html.Parse (log) where
+
+import Control.Applicative
+import qualified Data.Attoparsec.ByteString as AB
+import qualified Data.ByteString.Lazy as BL
+import qualified Data.Text as T
+import qualified Data.Text.Lazy as TL
+import qualified Data.Text.Lazy.Encoding as TLE
+import qualified Df1 as D
+import qualified Xmlbf as X
+import Prelude hiding (log)
+
+-- | An "Xmlbf" parser for a 'D.Log' rendered as HTML as 'Df1.Html.Render.log' renders it.
+-- 
+-- Notice that this parser will not ignore leading and trailing white space in the HTML.
+-- It will become part of the parsed 'D.Key', 'D.Value', 'D.Segment', 'D.Message'.
+log :: X.Parser D.Log
+log = X.pElement "div" $ do
+  attrClass "df1-log"
+  t <- parseTime
+  p <- parsePaths
+  l <- parseLevel
+  m <- parseMessage
+  let raw = BL.toStrict $ TLE.encodeUtf8 $ TL.intercalate " " [t, p, l, m]
+  case AB.parseOnly D.parse raw of
+    Left _ -> fail "Could not parse Log."
+    Right a -> pure a
+
+attrClass :: T.Text -> X.Parser ()
+attrClass t = do
+  attrs <- X.pAttr "class"
+  case elem t (T.words attrs) of
+    False -> fail ("Expected \"class\" value to contain " <> show t <> ".")
+    True -> pure ()
+
+parseTime :: X.Parser TL.Text
+parseTime = X.pElement "span" $ do 
+  attrClass "df1-time"
+  X.pText <|> pure ""
```

Comment:
This shouldn't fallback to succeeding with empty text. Contrary to things like `Message`, the timestamp can't be empty.